src/tike/ptycho/solvers/rpie.py

```diff
@@ -162,14 +163,16 @@ def rpie(
     return parameters
 
 
-def _update_wavefront(data, varying_probe, scan, psi, op=None):
+def _update_wavefront(data, varying_probe, scan, psi, position_options, op=None,):
 
     farplane = op.fwd(probe=varying_probe, scan=scan, psi=psi)
     intensity = cp.sum(
         cp.square(cp.abs(farplane)),
         axis=list(range(1, farplane.ndim - 2)),
     )
     cost = getattr(objective, f'{op.propagation.model}_each_pattern')(data, intensity)
+    if position_options is not None:
+        position_options.confidence[..., 0] = cost
```

Comment:
This is an appropriate place for storing this info, right?

Reply:
This seems like a reasonable place to store the confidence values.